.github/workflows/muelu_clang_format.yml

```diff
@@ -0,0 +1,18 @@
+name: Check MueLu clang-format
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.16.2
+      with:
+        source: './packages/muelu/src'
+        exclude: './lib'
```

Comment:
What's that exclusion for?

Reply:
So I don't forget that's an option in case we need it. I can remove it if you'd like

Reply:
Actually, we'll need to exclude the master list, so now I have a good use for it :)

Reply:
Why? The master list can be formatted correctly.

Reply:
I think it would require syncing formatting between the xml and cpp for the MasterList in a weird way. Test results are below

<details>
<summary> Failing: MueLu_MasterList </summary>

```
abstract.tex matches
acknowledgment.tex matches
appendix.tex matches
definitions.tex matches
introduction.tex matches
license.tex matches
mloptions_aggregation.tex matches
mloptions_general.tex matches
mloptions_misc.tex matches
mloptions_multigrid.tex matches
mloptions_rebalancing.tex matches
mloptions_refmaxwell.tex matches
mloptions_reuse.tex matches
mloptions_smoothing_and_coarse.tex matches
mloptions.tex matches
mueluguide.tex matches
muemex.tex matches
multigrid.tex matches
options_aggregation.tex matches
options_general.tex matches
options_misc.tex matches
options_multigrid.tex matches
options_rebalancing.tex matches
options_refmaxwell.tex matches
options_reuse.tex matches
options_smoothing_and_coarse.tex matches
options.tex matches
paramlist_hidden.tex matches
paramlist.tex matches
performance.tex matches
references.tex matches
SANDdistribution.tex matches
starting.tex matches
62d61
< 
76d74
< 
96,100c94,107
<       std::stringstream temp1; temp1 << "\"" << "MGV" << "\"";
<       std::stringstream temp2; temp2 << "\"" << "MGV" << "\"";
<       if (value == temp1.str() ) { ss << "<Parameter name=\"cycle type\" type=\"string\" value=\"V\"/>"; }
<       else if (value == temp2.str()) { ss << "<Parameter name=\"cycle type\" type=\"string\" value=\"W\"/>"; }
<       else TEUCHOS_TEST_FOR_EXCEPTION(true, MueLu::Exceptions::RuntimeError, "MasterList::interpretParameterName, Line " << __LINE__ << ". "
---
>     std::stringstream temp1;
>     temp1 << "\""
>           << "MGV"
>           << "\"";
>     std::stringstream temp2;
>     temp2 << "\""
>           << "MGV"
>           << "\"";
>     if (value == temp1.str()) {
>       ss << "<Parameter name=\"cycle type\" type=\"string\" value=\"V\"/>";
>     } else if (value == temp2.str()) {
>       ss << "<Parameter name=\"cycle type\" type=\"string\" value=\"W\"/>";
>     } else
>       TEUCHOS_TEST_FOR_EXCEPTION(true, MueLu::Exceptions::RuntimeError, "MasterList::interpretParameterName, Line " << __LINE__ << ". "
107,108c114,121
<       std::stringstream temp; temp << "\"" << "1" << "\"";
<       if (value == temp.str() ) { ss << "<Parameter name=\"multigrid algorithm\" type=\"string\" value=\"pg\"/>"; return ss.str(); }
---
>     std::stringstream temp;
>     temp << "\""
>          << "1"
>          << "\"";
>     if (value == temp.str()) {
>       ss << "<Parameter name=\"multigrid algorithm\" type=\"string\" value=\"pg\"/>";
>       return ss.str();
>     }
112c125,128
<       std::stringstream temp1; temp1 << "\"" << "1" << "\"";
---
>     std::stringstream temp1;
>     temp1 << "\""
>           << "1"
>           << "\"";
121,171c137,336
< 
<     if (name == "output filename") { ss << "<Parameter name=\"output filename\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "number of equations") { ss << "<Parameter name=\"number of equations\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "max levels") { ss << "<Parameter name=\"max levels\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "W cycle start level") { ss << "<Parameter name=\"W cycle start level\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "coarse grid correction scaling factor") { ss << "<Parameter name=\"coarse grid correction scaling factor\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "fuse prolongation and update") { ss << "<Parameter name=\"fuse prolongation and update\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "number of vectors") { ss << "<Parameter name=\"number of vectors\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "problem: symmetric") { ss << "<Parameter name=\"problem: symmetric\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "hierarchy label") { ss << "<Parameter name=\"hierarchy label\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "aggregation: drop tol") { ss << "<Parameter name=\"aggregation: drop tol\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "print initial parameters") { ss << "<Parameter name=\"print initial parameters\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "print unused parameters") { ss << "<Parameter name=\"print unused parameters\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: damping factor") { ss << "<Parameter name=\"sa: damping factor\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: use filtered matrix") { ss << "<Parameter name=\"sa: use filtered matrix\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: eigen-analysis type") { ss << "<Parameter name=\"sa: eigen-analysis type\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: eigenvalue estimate num iterations") { ss << "<Parameter name=\"sa: eigenvalue estimate num iterations\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: use rowsumabs diagonal scaling") { ss << "<Parameter name=\"sa: use rowsumabs diagonal scaling\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: enforce constraints") { ss << "<Parameter name=\"sa: enforce constraints\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: max eigenvalue") { ss << "<Parameter name=\"sa: max eigenvalue\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: rowsumabs diagonal replacement tolerance") { ss << "<Parameter name=\"sa: rowsumabs diagonal replacement tolerance\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: rowsumabs use automatic diagonal tolerance") { ss << "<Parameter name=\"sa: rowsumabs use automatic diagonal tolerance\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: rowsumabs diagonal replacement value") { ss << "<Parameter name=\"sa: rowsumabs diagonal replacement value\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "sa: rowsumabs replace single entry row with zero") { ss << "<Parameter name=\"sa: rowsumabs replace single entry row with zero\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "replicate: npdes") { ss << "<Parameter name=\"replicate: npdes\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "combine: numBlks") { ss << "<Parameter name=\"combine: numBlks\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "pcoarsen: element") { ss << "<Parameter name=\"pcoarsen: element\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "pcoarsen: schedule") { ss << "<Parameter name=\"pcoarsen: schedule\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "pcoarsen: hi basis") { ss << "<Parameter name=\"pcoarsen: hi basis\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "pcoarsen: lo basis") { ss << "<Parameter name=\"pcoarsen: lo basis\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "smoother: neighborhood type") { ss << "<Parameter name=\"smoother: neighborhood type\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "tentative: calculate qr") { ss << "<Parameter name=\"tentative: calculate qr\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "tentative: constant column sums") { ss << "<Parameter name=\"tentative: constant column sums\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: enable") { ss << "<Parameter name=\"repartition: enable\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: start level") { ss << "<Parameter name=\"repartition: start level\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: use map") { ss << "<Parameter name=\"repartition: use map\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: use subcommunicators in place") { ss << "<Parameter name=\"repartition: use subcommunicators in place\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: node repartition level") { ss << "<Parameter name=\"repartition: node repartition level\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: node id") { ss << "<Parameter name=\"repartition: node id\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: min rows per proc") { ss << "<Parameter name=\"repartition: min rows per proc\" type=\"int\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "repartition: max imbalance") { ss << "<Parameter name=\"repartition: max imbalance\" type=\"double\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "use external multigrid package") { ss << "<Parameter name=\"use external multigrid package\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "maxwell1: dump matrices") { ss << "<Parameter name=\"maxwell1: dump matrices\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: mode") { ss << "<Parameter name=\"refmaxwell: mode\" type=\"string\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: disable addon") { ss << "<Parameter name=\"refmaxwell: disable addon\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: use as preconditioner") { ss << "<Parameter name=\"refmaxwell: use as preconditioner\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: dump matrices") { ss << "<Parameter name=\"refmaxwell: dump matrices\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: subsolves on subcommunicators") { ss << "<Parameter name=\"refmaxwell: subsolves on subcommunicators\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: enable reuse") { ss << "<Parameter name=\"refmaxwell: enable reuse\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: skip first (1,1) level") { ss << "<Parameter name=\"refmaxwell: skip first (1,1) level\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
<     if (name == "refmaxwell: normalize nullspace") { ss << "<Parameter name=\"refmaxwell: normalize nullspace\" type=\"bool\" value=" << value << "/>"; return ss.str(); }      
---
>   if (name == "output filename") {
>     ss << "<Parameter name=\"output filename\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "number of equations") {
>     ss << "<Parameter name=\"number of equations\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "max levels") {
>     ss << "<Parameter name=\"max levels\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "W cycle start level") {
>     ss << "<Parameter name=\"W cycle start level\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "coarse grid correction scaling factor") {
>     ss << "<Parameter name=\"coarse grid correction scaling factor\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "fuse prolongation and update") {
>     ss << "<Parameter name=\"fuse prolongation and update\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "number of vectors") {
>     ss << "<Parameter name=\"number of vectors\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "problem: symmetric") {
>     ss << "<Parameter name=\"problem: symmetric\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "hierarchy label") {
>     ss << "<Parameter name=\"hierarchy label\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "aggregation: drop tol") {
>     ss << "<Parameter name=\"aggregation: drop tol\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "print initial parameters") {
>     ss << "<Parameter name=\"print initial parameters\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "print unused parameters") {
>     ss << "<Parameter name=\"print unused parameters\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: damping factor") {
>     ss << "<Parameter name=\"sa: damping factor\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: use filtered matrix") {
>     ss << "<Parameter name=\"sa: use filtered matrix\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: eigen-analysis type") {
>     ss << "<Parameter name=\"sa: eigen-analysis type\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: eigenvalue estimate num iterations") {
>     ss << "<Parameter name=\"sa: eigenvalue estimate num iterations\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: use rowsumabs diagonal scaling") {
>     ss << "<Parameter name=\"sa: use rowsumabs diagonal scaling\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: enforce constraints") {
>     ss << "<Parameter name=\"sa: enforce constraints\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: max eigenvalue") {
>     ss << "<Parameter name=\"sa: max eigenvalue\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: rowsumabs diagonal replacement tolerance") {
>     ss << "<Parameter name=\"sa: rowsumabs diagonal replacement tolerance\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: rowsumabs use automatic diagonal tolerance") {
>     ss << "<Parameter name=\"sa: rowsumabs use automatic diagonal tolerance\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: rowsumabs diagonal replacement value") {
>     ss << "<Parameter name=\"sa: rowsumabs diagonal replacement value\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "sa: rowsumabs replace single entry row with zero") {
>     ss << "<Parameter name=\"sa: rowsumabs replace single entry row with zero\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "replicate: npdes") {
>     ss << "<Parameter name=\"replicate: npdes\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "combine: numBlks") {
>     ss << "<Parameter name=\"combine: numBlks\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "pcoarsen: element") {
>     ss << "<Parameter name=\"pcoarsen: element\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "pcoarsen: schedule") {
>     ss << "<Parameter name=\"pcoarsen: schedule\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "pcoarsen: hi basis") {
>     ss << "<Parameter name=\"pcoarsen: hi basis\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "pcoarsen: lo basis") {
>     ss << "<Parameter name=\"pcoarsen: lo basis\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "smoother: neighborhood type") {
>     ss << "<Parameter name=\"smoother: neighborhood type\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "tentative: calculate qr") {
>     ss << "<Parameter name=\"tentative: calculate qr\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "tentative: constant column sums") {
>     ss << "<Parameter name=\"tentative: constant column sums\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: enable") {
>     ss << "<Parameter name=\"repartition: enable\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: start level") {
>     ss << "<Parameter name=\"repartition: start level\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: use map") {
>     ss << "<Parameter name=\"repartition: use map\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: use subcommunicators in place") {
>     ss << "<Parameter name=\"repartition: use subcommunicators in place\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: node repartition level") {
>     ss << "<Parameter name=\"repartition: node repartition level\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: node id") {
>     ss << "<Parameter name=\"repartition: node id\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: min rows per proc") {
>     ss << "<Parameter name=\"repartition: min rows per proc\" type=\"int\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "repartition: max imbalance") {
>     ss << "<Parameter name=\"repartition: max imbalance\" type=\"double\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "use external multigrid package") {
>     ss << "<Parameter name=\"use external multigrid package\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "maxwell1: dump matrices") {
>     ss << "<Parameter name=\"maxwell1: dump matrices\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: mode") {
>     ss << "<Parameter name=\"refmaxwell: mode\" type=\"string\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: disable addon") {
>     ss << "<Parameter name=\"refmaxwell: disable addon\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: use as preconditioner") {
>     ss << "<Parameter name=\"refmaxwell: use as preconditioner\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: dump matrices") {
>     ss << "<Parameter name=\"refmaxwell: dump matrices\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: subsolves on subcommunicators") {
>     ss << "<Parameter name=\"refmaxwell: subsolves on subcommunicators\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: enable reuse") {
>     ss << "<Parameter name=\"refmaxwell: enable reuse\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: skip first (1,1) level") {
>     ss << "<Parameter name=\"refmaxwell: skip first (1,1) level\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
>   if (name == "refmaxwell: normalize nullspace") {
>     ss << "<Parameter name=\"refmaxwell: normalize nullspace\" type=\"bool\" value=" << value << "/>";
>     return ss.str();
>   }
395,398c560,561
< "</ParameterList>"
< ;
<   std::map<std::string,std::string> MasterList::DefaultProblemTypeLists_ = DefaultProblemStrings<std::string,std::string>
< ("Poisson-2D",
---
>     "</ParameterList>";
> std::map<std::string, std::string> MasterList::DefaultProblemTypeLists_ = DefaultProblemStrings<std::string, std::string>("Poisson-2D",
408,410c571
<     "</ParameterList>"
<   )
< ("Poisson-2D-complex",
---
>                                                                                                                           "</ParameterList>")("Poisson-2D-complex",
426,428c587
<     "</ParameterList>"
<   )
< ("Poisson-3D",
---
>                                                                                                                                               "</ParameterList>")("Poisson-3D",
438,440c597
<     "</ParameterList>"
<   )
< ("Poisson-3D-complex",
---
>                                                                                                                                                                   "</ParameterList>")("Poisson-3D-complex",
456,458c613
<     "</ParameterList>"
<   )
< ("Elasticity-2D",
---
>                                                                                                                                                                                       "</ParameterList>")("Elasticity-2D",
468,470c623
<     "</ParameterList>"
<   )
< ("Elasticity-2D-complex",
---
>                                                                                                                                                                                                           "</ParameterList>")("Elasticity-2D-complex",
486,488c639
<     "</ParameterList>"
<   )
< ("Elasticity-3D",
---
>                                                                                                                                                                                                                               "</ParameterList>")("Elasticity-3D",
498,500c649
<     "</ParameterList>"
<   )
< ("Elasticity-3D-complex",
---
>                                                                                                                                                                                                                                                   "</ParameterList>")("Elasticity-3D-complex",
516,518c665
<     "</ParameterList>"
<   )
< ("MHD",
---
>                                                                                                                                                                                                                                                                       "</ParameterList>")("MHD",
552,554c699
<     "</ParameterList>"
<   )
< ("ConvectionDiffusion",
---
>                                                                                                                                                                                                                                                                                           "</ParameterList>")("ConvectionDiffusion",
574,576c719
<     "</ParameterList>"
<   )
< ;
---
>                                                                                                                                                                                                                                                                                                               "</ParameterList>");
1007,1010c1150
<          ("refmaxwell: normalize nullspace","refmaxwell: normalize nullspace")
<       ;
< 
< }
---
>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             ("refmaxwell: normalize nullspace", "refmaxwell: normalize nullspace");
1011a1152
> }  // namespace MueLu
MueLu_MasterList.cpp does not match
Do not edit MueLu_MasterList.cpp directly, but edit masterList.xml and run update_params.sh
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[6952,1],0]
  Exit code:    1
--------------------------------------------------------------------------
```
</details>

Reply:
You could just take what clang format tells you and apply that to the template that is used to generate the master list.

Reply:
I just tried it out, and it's easy enough to fix the header part of the template, but the part of the MasterList that is handled by
`xsltproc gen_interpreter.xsl masterList.xml >> $code_file` has a very specific format it generates, and it conflicts with what clang-format wants.